### PR TITLE
Check connection

### PR DESCRIPTION
**Pull request by BIG-RAT**
Add feedback to the user when check connection fails (#35).  For example:
Screen Shot 2020-09-11 at 6 53 58 AM

### CODE REVIEW

File: Source/Model/Alert.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    func display(header: String, message: String) {
+        DispatchQueue.main.async {
+            let dialog: NSAlert = NSAlert()
+            dialog.messageText = header
+            dialog.informativeText = message
+            dialog.alertStyle = NSAlert.Style.warning
+            dialog.addButton(withTitle: "OK")
+            dialog.runModal()
+        }
+    }   // func alert_dialog - end
```
Comment (kkot) [inline Source/Model/Alert.swift:40]: How is this comment useful?

**Review by kkot — DISMISSED** [review]
The changes look good, thank you for doing them. I added some just two small comments.

BTW for a person that will merge. I also tested it and it works nicely.

File: Source/Model/Alert.swift (hunk 1 above)
Comment (BIG-RAT) [reply Source/Model/Alert.swift:40]: Probably not too helpful here.  Just a habit I got into that helped me identify the end of functions that contained many more lines.

Comment (watkyn) [reply Source/Model/Alert.swift:40]: I would leave that off as well, hoping we won't have functions that don't fit on one screen.

File: Source/Model/Alert.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
```
Comment (watkyn) [inline Source/Model/Alert.swift:40]:
```suggestion
    }
```

**Review by watkyn — APPROVED** [review]

**Review by macblazer — APPROVED** [review]
Does what it says.  I'm not a fan of the two call site locations instead of putting this into the `handleCheckConnectionFailure` method, but it works acceptably well.

Maybe a future refactor on this code can remove the `Optional` from the `Bool` parameter on `handleCheckConnectionFailure` in favor of a default of `false`, and add an alert message String parameter in order to move the alert call into this error handling function.  That could also remove the need for the `DispatchQueue.main.async` call in the Alert class because it's already on the main thread.

**Pull request merged**